### DISCUSSION TITLE
新增 Beamer 样式和工程目录结构调整的咨询

### DISCUSSION THREAD

**Discussion by obster-y** [obster-y/XJTU-thesis#106] [Ideas]
准备新增 beamer 相关的 .sty，再根据之前的 PR 添加 .vscode/settings.json 直接写好编译选项给 LaTeX Workshop 使用

但是之前变过一次整体工程目录，是因为加了 Office 模板，后来项目拆出去了导致现在目录结构多一层 LaTeX_Template/，在考虑把这层目录去掉

咨询一下两位是否有什么想法 @ZhangSongyi @sikouhjw

**Comment by ZhangSongyi** [comment]
可以不调整目录结构，直接用vscode的workspace功能来实现（参见[这里](https://github.com/microsoft/vscode/issues/15909)的讨论）

我在 [这里](https://github.com/ZhangSongyi/XJTU-thesis/commit/5efe6a6fcdae5f1dbff74bb24d4e5b7d051ce2c4) 做了个测试
vscode选择`文件`-`从文件打开工作区`-选择`.code-workspace`文件就行

**Comment by ZhangSongyi** [comment]
这两天又试了几个新的功能（参见 [这个commit](https://github.com/ZhangSongyi/XJTU-thesis/commit/574767d32c86aa814456ee908199928d332f8a2f)）

- 加了一个专门存放输出pdf的目录，并且在recipes里加了个拷贝指令
![image](https://github.com/obster-y/XJTU-thesis/assets/12276998/720a1ba4-34ea-4889-bc85-e8203d030b45)
- 加了一个Task，用来调用清理的脚本，图形界面里可以从`终端`-`运行任务`里唤起

总之仅供参考，不过如果你要直接合的话我可以创建个PR

**Reply by obster-y** [reply]
>     * 加了一个专门存放输出pdf的目录，并且在recipes里加了个拷贝指令

可以来一个 PR

>     * 加了一个Task，用来调用清理的脚本，图形界面里可以从`终端`-`运行任务`里唤起

我准备去掉两个 clear.*，直接写到 latexmkrc 和 settings.json 里，以后让用户用 latexmkrc -c 或者直接点清理了

**Comment by obster-y** [comment]
@lzl32947 您之前 fork 后的仓库不见了，里面有很多建议很好，现在准备更新，不知道能不能再开放一下仓库？

**Reply by lzl32947** [reply]
您好，因为仓库主要是在毕业论文答辩之前完成的，所以主要是在原模板上进行的微小更新，由于我个人对Latex的了解并不多，因此大部分更改比较类似于救急变更，并没有考虑到编译后的整体协调性，鉴于未来可能很少使用Latex，因故我在毕业论文编译完成后就已经删除了原仓库的内容。不过我保留了本地仓库，现将我个人进行微调后的cls及Readme.md文件附在本Discussion中。

[XJTU-thesis-CS.zip](https://github.com/obster-y/XJTU-thesis/files/11550539/XJTU-thesis-CS.zip)

在实际调整中部分格式的调整很难与word中达到完全一致的效果，在我个人看来问题可能存在以下几点：

- 浮动体上下间距很难固定与latex一致，表现在表格、图片前后段落与正文间高度。原word模板中表格与断后文字间隔太小，不知是否需要调整。
- 图表的标题间距很难固定，在存在子图时很难调节，结合上一条，在部分页面中caption的间距会有一定的变化。
- ```xtabular```能够创建续表，但是在和```tabular```和```tabularx```合用时，如果页面中已经出现了浮动体，那么```xtabular```的间距就会有变化。
- 在word模板中，表格第一行，即表头的高度普遍偏小。

这是之前我在调节时遇到的一些显著问题，由于本人接触latex并不多，因此很难对其进行整体上的修改。

非常高兴能帮上您的忙。

**Reply by obster-y** [reply]
感谢您的分享☻

**Comment by ZhangSongyi** [comment]
@obster-y 独立Build目录以后，可以把`.gitignore`里的`*.pdf`去掉，换成`Build\*.pdf`
要不然的话pdf形式的图片不会被git追踪

**Reply by ZhangSongyi** [reply]
顺便可以再挂三行

    !.vscode
    .vscode/*
    !.vscode/settings.json

以避免追踪`.vscode`目录里的杂项文件

**Reply by obster-y** [reply]
> @obster-y 独立Build目录以后，可以把`.gitignore`里的`*.pdf`去掉，换成`Build\*.pdf` 要不然的话pdf形式的图片不会被git追踪

我忘了，之前是不准备让git跟踪二进制文件的，包括图片和pdf，但是刚看 gitignore 没有写图片，等下加上

**Comment by ZhangSongyi** [comment]
latexmkrc 里面的指令最好和vscode setting同步一下

可以把[这里](https://github.com/obster-y/XJTU-thesis/blob/fe2dced571521be2095def0a57dd4acfa7dcaab7/latexmkrc#L11) 改成
`$xelatex = "xelatex -synctex=1 -interaction=nonstopmode -file-line-error -shell-escape %O %S";`
（主要是加上`-file-line-error`一条，这样可以在vscode的错误面板里显示具体发生问题的位置）
